src/dav.js

```diff
@@ -0,0 +1,372 @@
+const { createClient } = require('webdav')
+const parser = require('./xmlParser.js')
+
+export class Dav {
+  constructor (baseUrl, baseUrlv2) {
+    this._XML_CHAR_MAP = {
+      '<': '&lt;',
+      '>': '&gt;',
+      '&': '&amp;',
+      '"': '&quot;',
+      "'": '&apos;'
+    }
+
+    this.baseUrl = baseUrl
+
+    this.userName = null
+
+    this.password = null
+
+    this.xmlNamespaces = {
+      'DAV:': 'd',
+      'http://owncloud.org/ns': 'oc'
+    }
+
+    this.xmlNamespacesComponents = {
+      d: 'DAV:',
+      oc: 'http://owncloud.org/ns'
+    }
+
+    this.client = createClient(baseUrl, {})
+    this.clientv2 = createClient(baseUrlv2, {})
+  }
+
+  _escapeXml (s) {
+    return s.replace(/[<>&"']/g, function (ch) {
+      return this._XML_CHAR_MAP[ch]
+    })
+  };
+
+  /**
+   * Generates a propFind request.
+   *
+   * @param {string} url Url to do the propfind request on
+   * @param {Array} properties List of properties to retrieve.
+   * @param {string} depth "0", "1" or "infinity"
+   * @param {Object} [headers] headers
+   * @return {Promise}
+   */
+  propFind (path, properties, depth = '0', headers = {}, options = {}) {
+    // depth header must be a string, in case a number was passed in
+    depth = '' + depth
+
+    headers = headers || {}
+
+    headers.Depth = depth
+    headers['Content-Type'] = 'application/xml; charset=utf-8'
```

Comment:
You had a constant for the tests with `'application/xml; charset=utf-8'` as content-type headers. Could you move that to another location and reuse it in this class as well (there are other occurrences in this class).